pkgs/development/compilers/elm/default.nix

```diff
@@ -30,12 +39,25 @@ let
 
             /*
             The elm-format expression is updated via a script in the https://github.com/avh4/elm-format repo:
-            `pacakge/nix/build.sh`
+            `package/nix/build.sh`
             */
             elm-format = justStaticExecutables (doJailbreak (self.callPackage ./packages/elm-format.nix {}));
 
+            elmi-to-json = justStaticExecutables (self.callPackage ./packages/elmi-to-json.nix {});
+
             inherit fetchElmDeps;
             elmVersion = elmPkgs.elm.version;
+
+            /*
+            Node/NPM based dependecies can be upgraded using script
+            `packages/generate-node-packages.sh`.
+            Packages which rely on `bin-wrap` will fail by default
+            and can be patched using `patchBinwrap` function defined in `packages/patch-binwrap.nix`.
+            */
+            elm-test = patchBinwrap [elmi-to-json] elmNodePackages.elm-test;
+            elm-verify-examples = patchBinwrap [elmi-to-json] elmNodePackages.elm-verify-examples;
```

Comment:
Wouldn't it be better to have elm-test and elm-verify-examples use patches on their own source code to remove the dependency on the npm elmi-to-json, instead of having them depend on a elmi-to-json with a patched binwrap?

Reply:
well, it would be possible for sure. In the end, it's a sort of tradeoff. This solution is likely to be cheaper to maintain going forward I guess.

Reply:
feel free to submit suggestion as I'm not entirely sure if I understand you correctly.